src/rules/extensions.js

```diff
@@ -46,9 +46,10 @@ module.exports = {
       typeof configuration === 'object' ? configuration : context.options[1]
     )
 
-    function isUseOfExtensionRequired(extension) {
+    function isUseOfExtensionRequired(extension, isModule) {
       if (!has(modifiers, extension)) { modifiers[extension] = defaultConfig }
-      return modifiers[extension] === 'always'
+      const modifier = modifiers[extension]
+      return (!isModule && modifier === 'ignorePackages') || modifier === 'always'
```

Comment:
it should not be ignored just because it's an external package, rather only when it's the "main" of an external package.

In other words, `require('foo')` and `require('@foo/bar')` should be ignored, but subpaths under these should not.